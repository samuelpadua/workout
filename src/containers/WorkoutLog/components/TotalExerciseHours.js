import React from 'react'
import PropTypes from 'prop-types'
import { Flex } from '@rebass/grid'
import styled from 'styled-components'
import { colors } from '../../../constants'


const MessageStyled = styled.h3`
  color: ${colors.success};
`

function TotalExerciseHours(props) {
  const { hours } = props
  return (
    <Flex p={20} justifyContent="center">
      <MessageStyled>
        { `${hours} horas de exerc√≠cios realizados` }
      </MessageStyled>
    </Flex>
  )
}

TotalExerciseHours.propTypes = {
  hours: PropTypes.number,
}

TotalExerciseHours.defaultProps = {
  hours: 0,
}

export default TotalExerciseHours
