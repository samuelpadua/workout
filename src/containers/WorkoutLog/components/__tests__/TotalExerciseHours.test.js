import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import TotalExerciseHours from '../TotalExerciseHours'

describe('TotalExerciseHours', () => {
  it('should render message with total of hours', () => {
    const tree = renderer
      .create(<TotalExerciseHours hours={8.5} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
