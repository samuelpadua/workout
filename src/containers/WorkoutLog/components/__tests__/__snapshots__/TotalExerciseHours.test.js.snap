// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TotalExerciseHours should render message with total of hours 1`] = `
.c0 {
  box-sizing: border-box;
  padding: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  color: #00D604;
}

<div
  className="c0"
>
  <h3
    className="c1"
  >
    8.5 horas de exerc√≠cios realizados
  </h3>
</div>
`;
